### DISCUSSION TITLE
Is it possible to display the bubble menu only when the table is dragged (selected)?

### DISCUSSION THREAD

**Discussion by ryuhangyeong** [ueberdosis/tiptap#5689] [General Questions] [answered]
## Description

![image](https://github.com/user-attachments/assets/0974d824-d646-4193-afc3-0034df3ea489)

When the current table has focus. A bubble menu will appear.

Is there a way to make the bubble menu appear only when the table is dragged (selected)?

## relationship

#4675

**Comment by kart-c** [comment]
Sorry, I am little confused😅
Do you want the bubble menu to appear when table has focus? Or when it is being dragged using a [drag handle](https://tiptap.dev/docs/editor/extensions/functionality/drag-handle)?

Assuming the bubble menu is expected to show up when the table has focus, adding `editor.isActive('table')` in the `shouldShow` prop in `BubbleMenu`  should help.

**Reply by ryuhangyeong** [reply]
## Example 1

When text in a table is dragged.

A bubble menu appears where you can edit text size and change color.

![image](https://github.com/user-attachments/assets/c066af0f-c85a-42b6-9427-77a0a85d9b8e)

## Example 2

When table rows and columns are selected

Shows bubble menus related to tables, such as adding and merging tables

![image](https://github.com/user-attachments/assets/81c564dd-46e1-4701-a0a0-5f8f4c6f9118)

Is it possible to differentiate these so that an appropriate bubble menu appears?

**Reply by ryuhangyeong** [reply]
I've created a new discussion for more detailed explanation.

#5697

**Reply by nperez0111** [reply]
Please do not make a new discussion you are just spamming

**Reply by kart-c** [reply]
Oh, so you want to listen to cells being selected? I am not sure if tiptap provides a similar functionality.

I noticed that in [this example](https://tiptap.dev/docs/editor/extensions/nodes/table#page-title), tiptap does provide merge cell functionality, where you select two or more cells and click on the merge button.

I saw the `merge` functionality comes from `prosemirror-tables`, so [this particular function](https://github.com/ProseMirror/prosemirror-tables/blob/master/src/commands.ts#L406) would be a good starting point.

**Comment by ryuhangyeong** [comment, marked as answer]
## Self Answer

```tsx
<BubbleMenu
  editor={editor}
  tippyOptions={{ placement: "bottom" }}
  shouldShow={({ editor, state }) => {
    const { selection } = state;

    const isText = isTextSelection(selection);

    return !isText && editor?.isActive("table");
  }}
>
  table bubble menu
</BubbleMenu>
```
